kernel/src/actions/mod.rs

```diff
@@ -321,44 +331,84 @@ impl Protocol {
     }
 }
 
-// given unparsed `table_features`, parse and check if they are subset of `supported_features`
-pub(crate) fn ensure_supported_features<T>(
-    table_features: &[String],
-    supported_features: &HashSet<T>,
-) -> DeltaResult<()>
```

Comment:
another quick comment: I _think_ (hope?) this PR can just add `table_feature: &[T]` here then we should be set? if we rely on the fact that unknown features will never be a subset of our supported ones? or we could go the extra mile and detect unknown up front but not sure if that's worth the effort?